tests/test_config.py

```diff
@@ -439,7 +440,32 @@ def _mock_isfile(filename):
     assert hass.config.time_zone.zone == 'Europe/Copenhagen'
     assert len(hass.config.whitelist_external_dirs) == 2
     assert '/tmp' in hass.config.whitelist_external_dirs
-    assert hass.config.config_source == config_util.SOURCE_STORAGE
+    assert hass.config.config_source == SOURCE_STORAGE
+
+
+async def test_updating_configuration(hass, hass_storage):
+    """Test updating configuration stores the new configuration."""
+    core_data = {
+        'data': {
+            'elevation': 10,
+            'latitude': 55,
+            'location_name': 'Home',
+            'longitude': 13,
+            'time_zone': 'Europe/Copenhagen',
+            'unit_system': 'metric'
+            },
+        'key': 'homeassistant.core_config',
+        'version': 1
+    }
+    hass_storage["homeassistant.core_config"] = dict(core_data)
+    await config_util.async_process_ha_core_config(
+        hass, {'whitelist_external_dirs': '/tmp'})
+    await hass.config.update(latitude=50)
+
+    new_core_data = dict(core_data)
+    new_core_data['data']['latitude'] = 50
```

Comment:
`dict` on the previous line is not a deep copy of the dictionary. This line is changing the original data.

Reply:
Fixed.